sweep/fee_bumper.go

```diff
@@ -518,6 +519,207 @@ type monitorRecord struct {
 	fee btcutil.Amount
 }
 
+// Start starts the publisher by subscribing to block epoch updates and kicking
+// off the monitor loop.
+func (t *TxPublisher) Start() error {
+	log.Info("TxPublisher starting...")
+
+	blockEvent, err := t.cfg.Notifier.RegisterBlockEpochNtfn(nil)
+	if err != nil {
+		return fmt.Errorf("register block epoch ntfn: %v", err)
+	}
+
+	select {
+	case bestBlock := <-blockEvent.Epochs:
+		t.currentHeight = bestBlock.Height
+
+	case <-t.quit:
+		log.Debugf("TxPublisher shutting down, exit start process")
+		return nil
+	}
+
+	t.wg.Add(1)
+	go t.monitor(blockEvent)
+
+	return nil
+}
+
+// Stop stops the publisher and waits for the monitor loop to exit.
+func (t *TxPublisher) Stop() {
+	log.Info("TxPublisher stopping...")
+
+	t.wg.Wait()
+	close(t.quit)
+}
+
+// monitor is the main loop driven by new blocks. Whevenr a new block arrives,
+// it will examine all the txns being monitored, and check if any of them needs
+// to be bumped. If so, it will attempt to bump the fee of the tx.
+func (t *TxPublisher) monitor(blockEvent *chainntnfs.BlockEpochEvent) {
+	defer blockEvent.Cancel()
+	defer t.wg.Done()
+
+	select {
+	case epoch, ok := <-blockEvent.Epochs:
+		if !ok {
+			// We should stop the publisher before stopping the
+			// chain service. Otherwise it indicates an error.
+			log.Error("Block epoch channel closed, exit monitor")
+
+			return
+		}
+
+		// Update the best known height for the publisher.
+		t.currentHeight = epoch.Height
+
+		// Check all monitored txns to see if any of them needs to be
+		// bumped.
+		t.processRecords()
+
+	case <-t.quit:
+		log.Debug("Fee bumper stopped, exit monitor")
+		return
+	}
+}
+
+// processRecords checks all the txns being monitored, and check if any of them
+// needs to be bumped. If so, it will attempt to bump the fee of the tx.
+func (t *TxPublisher) processRecords() {
+	// confirmed is a helper closure that returns a boolean indicating
+	// whether the tx is confirmed.
+	confirmed := func(txid chainhash.Hash,
+		event *chainntnfs.ConfirmationEvent) bool {
+
+		select {
+		case event := <-event.Confirmed:
+			log.Debugf("Sweep tx %v confirmed in block %v", txid,
```

Comment:
why not use ` r.tx.TxHash()` instead of the txid param ?

Reply:
replaced!